modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/pruning/PartitionPrunerImpl.java

```diff
@@ -71,6 +71,13 @@ public List<MappedFragment> apply(
                 continue;
             }
 
+            // Do not update colocation groups, in case of correlated fragment,
+            // because partitions for such fragments can be removed only at runtime.
+            if (fragment.correlated()) {
```

Comment:
this flag will be `false` in case of colocated join

Reply:
Thanks. Replaced with `correlated variables` check.